### ISSUE TITLE
Fail to record in Ubuntu with LTSP (thin client)

### ISSUE THREAD

**Issue by snadal** [MaartenBaert/ssr#456]
Hello,

I am trying to use ssr on an thin client running Ubuntu LTSP , but no matter the options I choose, I always get the same error.

[X11Input::InputThread] Error: Can't get image (using shared memory)!
Usually this means the recording area is not completely inside the screen. Or did you change the screen resolution?
[X11Input::InputThread] Exception 'X11Exception' in input thread.

I tried to run as root too, without luck.

Any hints ?

**Comment by vladimirus0** [comment]
Hi!
I have the same problem
You have solved this problem at the moment?

**Comment by StripedMonkey** [comment]
Both of you provide very few details :P
Are you recording the entire screen? A window? OpenGl?
What are you recording? A fullscreen application?
etc.

**Comment by vladimirus0** [comment]
window :)
i`m recording only the window of browser Google Chrome

**Comment by snadal** [comment]
I don't know if Vladimir case is the same as ours, but we are using LTSP, which uses XDMCP for thin clients, so it may be related with XDMCP way of rendering content on screen.

**Comment by MaartenBaert** [comment]
Many remote desktop protocols don't implement every feature of X11. In this case the missing feature is shared memory. This makes sense because shared memory can only be used for communication within the same computer, so it would have required special workarounds to make this work anyway.

I could add an option to use the old, slow protocol instead of shared memory, but this is already painfully inefficient even within the same computer, so if you try to run it over a network, even with gigabit ethernet, it will probably be almost unusable. Recording 1080p at 30 FPS would require a 2 Gbit/s connection, so you would be limited to 15 FPS even under ideal conditions.

Long story short, don't try to record through a remote desktop protocol. Either record completely client-side or completely server-side.

I know that it is possible to record inside a VNC session with SSR. This works perfectly because all the processing is done server-side. It's also possible to record the VNC window itself client-side, but the result is not as good because you will see the latency and compression of the VNC connection. I have never tried LTSP though.
